### ISSUE TITLE
sidechain ducking not working when using Jack Input Client as the sidechain source

### ISSUE THREAD

**Issue by wolftune** [obsproject/obs-studio#4335]
## Platform

Operating system and version: KDE Neon User Edition 5.21 (based on Ubuntu 20.04)
OBS Studio version: 26.1.1
OBS Log file: 
[2021-03-11 17-50-26.txt](https://github.com/obsproject/obs-studio/files/6127258/2021-03-11.17-50-26.txt)
[2021-03-11 17-25-34.txt](https://github.com/obsproject/obs-studio/files/6127259/2021-03-11.17-25-34.txt)

## Expected Behavior

I should be able to have two Jack input sources, one that has the compressor on is set to use the other one as sidechain source. The compresser should be working to only as the sidechain signal comes on.

## Current Behavior

I tried all the possible settings and nothing gets the compressor to respond to the sidechain. It acts as though no sidechain is selected.

## Steps to Reproduce

1. Add two Jack Input Clients and select different inputs to each from a Jack patchbay tool like qjackctl or Catia, verify they both show signals
2. Add a compressor to one of the clients and choose the other as the sidechain input
3. Play with the compressor options and the input signals for the sidechain
4. Notice that both the meters and the heard output indicate no sidechain ducking

**Comment by kkartaltepe** [comment]
Configuring jack isnt an obs issue.

**Comment by wolftune** [comment]
@kkartaltepe This bug is not at all about configuring Jack. I just didn't figure out yet if the issue is independent of Jack or not. I suppose I should do that test. But the point is that OBS has two audio sources, the OBS compressor is set, and the sidechain isn't working. All the sources are correctly going into OBS,  and the failure of the sidechain ducking is happening inside OBS *not* in Jack.

**Comment by wolftune** [comment]
@kkartaltepe I just verified that it is specific to Jack, but this has again nothing to do with Jack configuration. It's an OBS bug.

I successfully got the sidechain to work when using OBS audio inputs with media and pulseaudio sources. But when a Jack source is chosen as the sidechain source, the compression doesn't work. This is not because of Jack failing to send to OBS. OBS shows *and* puts through the audio signals just fine. So, OBS internally is receiving two audio inputs, it's just failing to use one of them as the sidechain source for some reason. Again, this is *after* Jack has successfully sent all the signals into OBS.

Again to summarize: I have now verified that the sidechain *works* with pulse or media audio sources. It seems to fail only with a Jack source.

**Comment by kkartaltepe** [comment]
Ducking is working with jack on my machine on master. Please check if that is also the case for you if you can compile, it may be related to timing changes to jack input that will be released in the next version.

**Comment by wolftune** [comment]
Okay, sorry for the confusion, I tried several tests, and now I got at least one time where it *is* working. If I can reliably reproduce this, I'll check back. Apologies for the hassle!

**Comment by wolftune** [comment]
FWIW, I tried again and now it's failing again. I tried several attempts to test if it related to multiple clients or which was set to what first. All I know at this time is I once had it working, it *was* ducking a Jack client from another Jack client. But most of the times including *after* that, I haven't gotten it to work. *Maybe* there's some particular combination or order of process that reliably will work? I don't know yet. Maybe the timing thing mentioned will be helpful.
